src/PluginsManager/Microsoft.Performance.Toolkit.PluginsManager.Core/AvailablePlugin.cs

```diff
@@ -2,62 +2,97 @@
+            this.pluginFetcher = fetcher;
         }
 
         /// <summary>
-        ///     Gets the identity of this plugin.
-        /// </summary>
-        public PluginIdentity Identity { get; }
-        
-        /// <summary>
-        ///     Gets the souce where this plugin is discovered.
+        ///     Gets the <see cref="AvailablePluginInfo"/> associated with this plugin.
         /// </summary>
-        public PluginSource PluginSource { get; }
+        public AvailablePluginInfo AvailablePluginInfo { get; }
 
-        /// <summary>
-        ///     Gets the human-readable name of this plugin.
-        /// </summary>
-        public string DisplayName { get; }
 
         /// <summary>
-        ///     Gets the user friendly description of this plugin.
-        /// </summary>   
-        public string Description { get; }
+        ///     Gets the metadata of this plugin.
+        /// </summary>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <returns>
+        ///     The metadata of a plugin.
+        /// </returns>
+        public Task<PluginMetadata> GetPluginMetadata(CancellationToken cancellationToken)
+        {
+            return this.pluginDiscoverer.GetPluginMetadataAsync(this.AvailablePluginInfo.Identity, cancellationToken);
+        }
 
         /// <summary>
-        ///     Gets the URI where the plugin package can be fetched.
+        ///     Gets all available versions of this plugin.
         /// </summary>
-        public Uri PluginPackageUri { get; }
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <returns>
+        ///      A collection of available plugins.
+        /// </returns>
+        public async Task<IReadOnlyCollection<AvailablePlugin>> GetAllVersions(CancellationToken cancellationToken)
+        {
+            IReadOnlyCollection<AvailablePluginInfo> pluginInfos = await this.pluginDiscoverer.DiscoverAllVersionsOfPlugin(
+                this.AvailablePluginInfo.Identity, cancellationToken);
+
+            return pluginInfos.Select(info => new AvailablePlugin(info, this.pluginDiscoverer, this.pluginFetcher)).ToList();
+        }
 
         /// <summary>
-        ///     Gets the Guid which identifies the platform (NuGet, Github etc.) the plugin package is hosted on.
+        /// 
         /// </summary>
-        public Guid FetcherTypeId { get; }
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <param name="progress">
+        ///     Indicates the progress of plugin installation.
+        /// </param>
+        /// <returns></returns>
+        public Task<Stream> GetPluginPackageStream(CancellationToken cancellationToken, IProgress<int> progress)
```

Comment:
this probably shouldn't be here... I imagine this method should be `public Task InstallAsync(CancellationToken cancellationToken, IProgress<int> progress)`

Maybe this should be an internal method?